### ISSUE TITLE
Config Parser Design

### ISSUE THREAD

**Issue by jmsalvador2395** [BDI-ML/ml-toolkit#1]
# main ideas
- [x] have configs in a directory structure.
- [x] build configs using built-in configparser package
- [x] design interface to prompt users for configs if none are provided

# parameters to account for
I've separated these parameters into categories for convenience. this structure doesn't necessarily have to be used for building configs

## general parameters
- [ ] task_name: the task to train on. this is used to identify what kind of model to train (ex. word2vec_cbow)
- [ ] device: the device to use for processing tensors (ex. cuda:0)
- [ ] seed: used to manually seed the random number generators for numpy and pytorch

## optimization parameters
- [ ] lr: the optimizer learning rate
- [ ] batch_size: the amount of examples to train on at a given time
- [ ] weight_decay: the regularization term to be used for the optimizer

## dataset specific parameters
- [ ] dataset_name: used to retrieve a dataset from huggingface
- [ ] data_dir: used to specify where the data is located (this is only used for custom datasets)
- [ ] shuffle: tells the program whether to shuffle the data or not
- [ ] num_epochs: the number of times to iterate over the training data
- [ ] cache_dir: where to store your dataset cache. (default would be in the project root)

## list of parameters used for word2vec (maybe just put them in a category called 'Model Parameters')
- [ ] embedding_dim: the dimensionality of the embeddings to train (ex. 128)
- [ ] window_size: defines the amount of context words to use when training word2vec (ex. 9)

## tools (this is for things like tokenizers and vocabularies)
- [ ] tokenizer: the desired tokenizer to use on the dataset
- [ ] min_freq: the minimum frequency a token needs to occur in the dataset before being indexed as '<UNK>'

**Comment by jmsalvador2395** [comment]
[link to the config parser api](https://docs.python.org/3/library/configparser.html)

**Comment by jmsalvador2395** [comment]
config search menu implemented

**Comment by jmsalvador2395** [comment]
note: implement use a class called `Trainer` to implement training tasks.

**Comment by jmsalvador2395** [comment]
Config parser design is done
